Sources/Web3Core/Structure/Block/Block.swift

```diff
@@ -102,6 +102,40 @@ extension Block: Decodable {
             return data
         }
     }
+
+    public func encode(to encoder: Encoder) throws {
+        var container = encoder.container(keyedBy: CodingKeys.self)
+        try container.encodeHex(number, forKey: .number)
+        try container.encodeHex(hash, forKey: .hash)
+        try container.encodeHex(parentHash, forKey: .parentHash)
+        try container.encodeHexIfPresent(nonce, forKey: .nonce)
+        try container.encodeHex(sha3Uncles, forKey: .sha3Uncles)
+        
+        try container.encodeHexIfPresent(logsBloom?.bytes, forKey: .logsBloom)
+
+        try container.encodeHex(transactionsRoot, forKey: .transactionsRoot)
+        try container.encodeHex(stateRoot, forKey: .stateRoot)
+        try container.encodeHex(receiptsRoot, forKey: .receiptsRoot)
+
+        try container.encodeIfPresent(miner?.address, forKey: .miner)
+
+        try container.encodeHex(difficulty, forKey: .difficulty)
+        try container.encodeHex(totalDifficulty, forKey: .totalDifficulty)
+        try container.encodeHex(extraData, forKey: .extraData)
+        try container.encodeHex(size, forKey: .size)
+        try container.encodeHex(gasLimit, forKey: .gasLimit)
+        try container.encodeHex(gasUsed, forKey: .gasUsed)
+
+        // optional, since pre EIP-1559 block haven't such property.
```

Comment:
Typo, please also fix that for the comment in `init(from decoder: Decoder)`.

It's not `... block haven't ...` but `... blocks didn't have ...`.